pkg/ctl/create/cluster.go

```diff
@@ -155,6 +156,10 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 		}
 	}
 
+	if err := validateZonesAndNodeZones(cmd, params); err != nil {
+		return err
+	}
```

Comment:
could this not be with the rest of the validation code & unit tested?

Reply:
I tried adding it there. But it's not called in time actually and not with the parameters I need.

Reply:
More precisely this `params.AvailabilityZones` is not available there.

Reply:
ahh sorry yeah validation is the wrong place (even though it probably should be the right place :upside_down_face: ). What about here https://github.com/weaveworks/eksctl/blob/df75054cf0b2696130f465ac7a0af7e4bbe7b851/pkg/ctl/cmdutils/configfile.go#L281-L336

we have a lot of flag validation logic there

Reply:
Yeah I tried that one too. But decided against it in the end. I don't remember why. It was a week ago :D

Reply:
But yeah, I put it here first. :)

Reply:
> Yeah I tried that one too. But decided against it in the end. I don't remember why. It was a week ago :D

:thunk: this definitely feels like a suitable place though? It has validation logic for when a user is using command line flags, e.g. https://github.com/weaveworks/eksctl/blob/main/pkg/ctl/cmdutils/configfile.go#L294 

> FYI, what you are showing is nodegroups. This is during cluster creation.

isn't what I linked for clusters? `func NewCreateClusterLoader(cmd *Cmd, ngFilter *filter.NodeGroupFilter, ng *api.NodeGroup, params *CreateClusterCmdParams) ClusterConfigLoader {`